editor/editor_file_system.cpp

```diff
@@ -1184,10 +1244,17 @@ void EditorFileSystem::scan_changes() {
 		scan_total = 0;
 		Thread::Settings s;
 		s.priority = Thread::PRIORITY_LOW;
+		// Accrues changes
 		thread_sources.start(_thread_func_sources, this, s);
+		// Because of multithreading, changes are applied in a check that occurs in the process loop.
+		// If the loop detects the multithread is done, then it applies the changes. See NOTIFICATION_PROCESS for more.
 	}
 }
 
+bool EditorFileSystem::_is_script_extendable_resource(const StringName &p_class) {
+	return !ClassDB::is_parent_class(p_class, Script::get_class_static()) && !ClassDB::is_parent_class(p_class, PackedScene::get_class_static()) && ClassDB::is_parent_class(p_class, Resource::get_class_static());
```

Comment:
`Resource` check should be at the beginning, as the other checks are useless if the classes aren't Resource.